src/clj/rems/application/model.clj

```diff
@@ -432,20 +441,19 @@
 
 (defn- enrich-resources [app-resources get-catalogue-item]
   (->> app-resources
-       (map :catalogue-item/id)
-       (map get-catalogue-item)
-       (map (fn [item]
-              {:catalogue-item/id (:id item)
-               :resource/id (:resource-id item)
-               :resource/ext-id (:resid item)
-               :catalogue-item/title (assoc (localization-for :title item)
-                                            :default (:title item))
-               ;; TODO: remove unused keys
-               :catalogue-item/start (:start item)
-               :catalogue-item/end (:end item)
-               :catalogue-item/enabled (:enabled item)
-               :catalogue-item/expired (:expired item)
-               :catalogue-item/archived (:archived item)}))
+       (map (fn [resource]
+              (merge
+               resource ; already contains :catalogue-item/id and :resource/ext-id
+               (let [item (get-catalogue-item (:catalogue-item/id resource))]
+                 {:resource/id (:resource-id item)
+                  :catalogue-item/title (assoc (localization-for :title item)
+                                               :default (:title item))
+                  ;; TODO: remove unused keys
+                  :catalogue-item/start (:start item)
+                  :catalogue-item/end (:end item)
+                  :catalogue-item/enabled (:enabled item)
+                  :catalogue-item/expired (:expired item)
+                  :catalogue-item/archived (:archived item)}))))
```

Comment:
I preferred the old version, it was more explicit about which keys are emitted.

Reply:
The problem is that in the tests `item` is `nil`, which results in even the known IDs being nilled.

Reply:
well how about still spelling out the keys instead of merging? that is,
```
{:catalogue-item/id (:catalogue-item/id resource)
 :resource/ext-id (:resource/ext-id resource)
 ...
 :catalogue-item/start (:start item)}

Reply:
👍